### PR TITLE
Extend statusline interface

### PR DESCRIPTION
**Pull request by petpetpetpet**
This is a non-breaking change to the external API. It adds one new externally facing function:
ale#statusline#FirstProblem(buffer, problem_type) that returns the first locline entry matching the input parameters.

Why do this? Because I wanted to get the line numbers of the first error, warning, style_error, style_warning that ale finds in the active buffer and it was harder to do this than it should have been.

Long story:

Basically I noticed that the people who wrote airline wanted to display the line numbers for the first artifact found by ale in their status line. In order to do that, they were grabbing the entire loclist, making a copy, running a filter on it by type, and finally using filteredloclist[0]['lnum']. All that for a line number. Insane.

### CODE REVIEW

File: doc/ale.txt [hunk 1]
```diff
@@ -3129,6 +3129,22 @@ ale#statusline#Count(buffer)                           *ale#statusline#Count()*
   `total`         -> The total number of problems.
 
 
+ale#statusline#FirstProblem(buffer, problem_type)
```
Comment (w0rp) [inline doc/ale.txt]: You can fit the doctag on the same line if you name the second argument `type`.

File: README.md [hunk 2]
```diff
@@ -576,8 +576,16 @@ let g:airline#extensions#ale#enabled = 1
...
 If you don't want to use vim-airline, you can implement your own statusline
-function without adding any other plugins. ALE provides a function for counting
-the number of problems for this purpose, named `ale#statusline#Count`.
+function without adding any other plugins. ALE provides some functions to 
+assist in this endeavour, including:
+
+* `ale#statusline#Count`: Which returns the number of problems found by ALE
+  for a specified buffer.
+* `ale#statusline#FirstProblem`: Which returns a dictionary containing the
+  full loclist details of the first problem of a specified type found by ale
```
Comment (w0rp) [inline README.md]:
```suggestion
  full loclist details of the first problem of a specified type found by ALE 
```

Still need to be changed.

Comment (petpetpetpet) [reply README.md]: sorry- i am now making the changes locally and will push them together.

**Comment by w0rp** [comment]
Cheers! :beers:
